lib/ansible/plugins/action/__init__.py

```diff
@@ -520,12 +535,48 @@ def _fixup_perms2(self, remote_paths, remote_user=None, execute=True):
                     if res['rc'] != 0:
                         raise AnsibleError('Failed to set file mode on remote temporary files (rc: {0}, err: {1})'.format(res['rc'], to_native(res['stderr'])))
 
-                res = self._remote_chown(remote_paths, self.get_become_option('become_user'))
-                if res['rc'] != 0 and remote_user in self._get_admin_users():
-                    # chown failed even if remote_user is administrator/root
-                    raise AnsibleError('Failed to change ownership of the temporary files Ansible needs to create despite connecting as a privileged user. '
-                                       'Unprivileged become user would be unable to read the file.')
-                elif res['rc'] != 0:
+                res = self._remote_chown(remote_paths, become_user)
+                if res['rc'] != 0:
+                    # First check if we are an admin/root user. If we are
+                    # and failed here, something weird has happened.
+                    if remote_user in self._get_admin_users():
+                        # chown failed even if remote_user is administrator/root
+                        raise AnsibleError('Failed to change ownership of the temporary files Ansible needs to create despite connecting as a privileged user. '
+                                           'Unprivileged become user would be unable to read the file.')
+
+                    # Otherwise, we're a normal user. We failed to chown the
+                    # paths to the unprivileged user, but if we have a common
+                    # group with them, we should be able to chown it to that.
+                    #
+                    # Note that we have no way of knowing if this will actually
+                    # work... just because chgrp exits successfully does not
+                    # mean that Ansible will work. We could check if the become
+                    # user is in the group, but this would create an extra
+                    # round trip.
```

Comment:
not really, we could append/chain the verification to the end of the same command string, but that is also 'fun' in certain contexts that will misreport, so we are better off just letting the final execution fail for permission issues